### DISCUSSION TITLE
[RFC] Input Semantic DOM Structure

### DISCUSSION THREAD

**Discussion by MadCcc** [ant-design/ant-design#41241] [RFCs - archive]
## Summary

In https://github.com/ant-design/ant-design/discussions/40221 we introduced semantic dom structure for components, and this time we would apply it to Input.

## API

> Since `classNames`, `styles` and `refs` have the same properties and different value types, we only use `classNames` to demonstrate.

**Input**
| Property | Description | 
| --- | --- |
| input | The `input` element itself |
| count | Count node |
| prefix | Prefix node |
| suffix | Suffix node |

**Input.TextArea**
| Property | Description | 
| --- | --- |
| textarea | The `textarea` element itself || count | Count node |
| count | Count node |

## Basic Example

```tsx
import React from 'react';
import { Input } from 'antd';

export default () => (
  <Input classNames={{ input: 'custom-input', affixWrapper: 'custom-wrapper' }} />
)
```

## How we teach this

In **feature** branch:

Add document for Input semantic dom structure.

**Comment by afc163** [comment]
count 是啥？

**Reply by MadCcc** [reply]
文字计数，这个算比较细的了，打星号的都是不确定会有的

**Comment by afc163** [comment]
affixWrapper 感觉比较怪。

**Reply by MadCcc** [reply]
先取了目前的 className。现在 group 被 Compact 取代了，直接叫 `wrapper` 也是一个方案。

**Reply by zombieJ** [reply]
> 先取了目前的 className。现在 group 被 Compact 取代了，直接叫 `wrapper` 也是一个方案。

不要叫 wrapper，这个过于贴近实现了

**Comment by arvinxx** [comment]
affixWrapper	-> container
